pkg/util/startuplogger/startuplogger.go

```diff
@@ -0,0 +1,81 @@
+package startuplogger
+
+import (
+	"encoding/json"
+	"os"
+	"time"
+
+	"k8s.io/klog/v2"
+)
+
+type ServiceData struct {
+	Name         string    `json:"name"`
+	Dependencies []string  `json:"dependencies"`
+	Start        time.Time `json:"start"`
+	Ready        time.Time `json:"ready"`
+}
+
+func (s ServiceData) MarshalJSON() ([]byte, error) {
+	type Alias ServiceData
+	return json.Marshal(struct {
+		Alias
+		StartupTime string `json:"duration"`
+	}{
+		Alias:       (Alias)(s),
+		StartupTime: s.Ready.Sub(s.Start).String(),
+	})
+}
+
+type StartupLogger struct {
+	services        []ServiceData
+	microshiftStart time.Time
+	microshiftReady time.Time
+}
+
+func (l StartupLogger) MarshalJSON() ([]byte, error) {
+	return json.Marshal(struct {
+		MicroshiftStart string        `json:"microshiftStart"`
+		MicroshiftReady string        `json:"microshiftReady"`
+		Services        []ServiceData `json:"services"`
+	}{
+		MicroshiftStart: l.microshiftStart.Format(time.RFC3339),
```

Comment:
time.MarshalJSON already uses time.RFC3339, so this is  conversion with intermediate struct is probably unnecessary